RadioController.py

```diff
@@ -1,43 +1,47 @@
+import math
+from typing import Dict, Any, Callable, Union, List, Tuple
 import struct
-from typing import Any, Callable, Dict, List, Union
 
 import SubpacketIDs
 from SubpacketIDs import SubpacketEnum
 
+import collections
+Header = collections.namedtuple('Header', ['subpacket_id', 'timestamp', 'header_length', 'data_length', 'total_length'])
+
 # CONSTANTS
 
-# TODO Review these based on spec
-# Map subpacket id (int) to length (int) in bytes. Only includes types with CONSTANT lengths
+# Map subpacket id to DATA length (excluding header) in bytes. Only includes types with CONSTANT lengths.
 PACKET_ID_TO_CONST_LENGTH: Dict[int, int] = {
-    SubpacketEnum.STATUS_PING.value: 4,
-    SubpacketEnum.EVENT.value: 2,
-    SubpacketEnum.GPS.value: 25,
-    SubpacketEnum.ACKNOWLEDGEMENT.value: 0000,  # TODO ack length?
-    SubpacketEnum.BULK_SENSOR.value: 42,
+    SubpacketEnum.STATUS_PING.value: 5,
+    SubpacketEnum.EVENT.value: 1,
+    SubpacketEnum.GPS.value: 24,
+    SubpacketEnum.ACKNOWLEDGEMENT.value: 0000,  # TODO ack length??
+    SubpacketEnum.BULK_SENSOR.value: 37,
 }
 for i in SubpacketIDs.get_list_of_sensor_IDs():
-    PACKET_ID_TO_CONST_LENGTH[i] = 5
+    PACKET_ID_TO_CONST_LENGTH[i] = 4
 
 # Check if packet of given type has constant length
 def isPacketLengthConst(subpacket_id):
     return subpacket_id in PACKET_ID_TO_CONST_LENGTH.keys()
 
 
-# TODO Update this when timestamp is going to be included in header
-# Map subpacket id to header size in bytes. Only includes types with CONSTANT lengths
-PACKET_ID_TO_HEADER_SIZE: Dict[int, int] = {
-    SubpacketEnum.STATUS_PING.value: 1,
-    SubpacketEnum.MESSAGE.value: 6,
-    SubpacketEnum.EVENT.value: 1,
-    SubpacketEnum.CONFIG.value: 1,
-    SubpacketEnum.GPS.value: 1,
-    SubpacketEnum.ACKNOWLEDGEMENT.value: 1,
-    SubpacketEnum.BULK_SENSOR.value: 1,
-}
-for i in SubpacketIDs.get_list_of_sensor_IDs():
-    PACKET_ID_TO_HEADER_SIZE[i] = 1
-
-
+# # TODO remove? since we have header helper that tracks header size. Would be useful, if header size varies more
+# # Map subpacket id to header size in bytes. Only includes types with CONSTANT lengths
+# PACKET_ID_TO_HEADER_SIZE: Dict[int, int] = {
+#     SubpacketEnum.STATUS_PING.value: 5,
+#     SubpacketEnum.MESSAGE.value: 6,
+#     SubpacketEnum.EVENT.value: 5,
+#     SubpacketEnum.CONFIG.value: 5,
+#     SubpacketEnum.GPS.value: 5,
+#     SubpacketEnum.ACKNOWLEDGEMENT.value: ,???
+#     SubpacketEnum.BULK_SENSOR.value: 5,
+# }
+# for i in SubpacketIDs.get_list_of_sensor_IDs():
+#     PACKET_ID_TO_HEADER_SIZE[i] = 1
```

Comment:
Let me know what you think, @jesseli2002 especially, regarding specifying header size somewhere or not